File: level1/javascript/두 정수 사이의 합 12912.js
function solution(a, b) {

    /* 방법 1
    if(a == b)
        return a;
    
    else if (a > b){
        let tmp = a;
        a = b;
        b = tmp;  
    }

    let answer = a;
    for(let i = a+1; i <= b; i++){
        console.log(answer, i);
        answer+= i;

    }
    
    return answer;
    */

   // 방법2
   return (Math.abs(a-b)+1)*(a+b) / 2;
}

console.log(solution(5,3));

File: level1/kotlin/두 정수 사이의 합 12912.kt
class Solution {
    fun solution(a: Int, b: Int): Long {
        val result = compareNum(a, b)
        var answer = 0L
        var value = result.first

        while (value <= result.second) {
            if (value < 0) answer -= value
            else answer += value
            value++
        }

        return answer
    }

    /** Pair<Int, Int> : first() -> min, second() -> max */
    fun compareNum(a: Int, b: Int): Pair<Int, Int> {
        return if (a < b) Pair(a, b) else Pair(b, a)
    }
}